utils/uf2conv.py

```diff
@@ -212,7 +212,7 @@ def write_file(name, buf):
 
 
 def load_families():
-    with open("uf2families.json") as f:
+    with open(os.path.join(sys.path[0], "uf2families.json")) as f:
```

Comment:
I think you need `os.path.dirname(os.path.abspath(__file__))` not `sys.path[0]` see https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory

Reply:
My search-fu found this one first :laughing: https://www.tutorialspoint.com/How-to-open-a-file-in-the-same-directory-as-a-Python-script

And I did test it out:

```
➜  uf2 git:(bcr-family-fixes) python3 utils/uf2conv.py --family aaa
Family ID needs to be a number or one of: ATMEGA32, SAML21, NRF52, ESP32, STM32L1, STM32L0, STM32WL, LPC55, STM32G0, GD32F350, STM32L5, STM32G4, MIMXRT10XX, STM32F7, SAMD51, STM32F4, FX2, STM32F2, STM32F1, STM32F0, SAMD21, STM32F3, STM32F407, STM32H7, STM32WB, ESP8266, KL32L2, STM32F407VG, NRF52840, ESP32S2, ESP32S3, ESP32C3, RP2040, STM32L4
➜  uf2 git:(bcr-family-fixes) ✗ cd utils    
➜  utils git:(bcr-family-fixes) ✗ python3 uf2conv.py --family aaa 
Family ID needs to be a number or one of: ATMEGA32, SAML21, NRF52, ESP32, STM32L1, STM32L0, STM32WL, LPC55, STM32G0, GD32F350, STM32L5, STM32G4, MIMXRT10XX, STM32F7, SAMD51, STM32F4, FX2, STM32F2, STM32F1, STM32F0, SAMD21, STM32F3, STM32F407, STM32H7, STM32WB, ESP8266, KL32L2, STM32F407VG, NRF52840, ESP32S2, ESP32S3, ESP32C3, RP2040, STM32L4
➜  utils git:(bcr-family-fixes) ✗ 
```

I like your answer better, so let me change it.

Reply:
Sorry about that! I run python interactively and sys.path[0] there is empty, but it does work the way you describe. You can leave as is or change it to "my" version (which is more explicit)

Reply:
Changed it to your version.